components/automate-cli/cmd/chef-automate/certRotate.go

```diff
@@ -0,0 +1,275 @@
+}{}
+
+var sshFlag = struct {
+	automate bool
+	postgres bool
+}{}
+
+var certRotateCmd = &cobra.Command{
+	Use:   "cert-rotate",
+	Short: "Chef Automate rotate cert",
+	Long:  "Chef Automate CLI command to rotate certificates",
+	RunE:  certRotate,
+}
+
+const (
+	FRONTEND_COMMANDS = `
+	sudo chef-automate config patch /tmp/%s;
+	export TIMESTAMP=$(date +'%s');
+	sudo mv /etc/chef-automate/config.toml /etc/chef-automate/config.toml.$TIMESTAMP;
+	sudo chef-automate config show > sudo /etc/chef-automate/config.toml`
+
+	BACKEND_COMMAND = `
+    sudo HAB_LICENSE=accept-no-persist hab config apply automate-ha-%s.default  $(date +"%s") /tmp/%s;`
+
+	dateFormat = "%Y%m%d%H%M%S"
+)
+
+func init() {
+	RootCmd.AddCommand(certRotateCmd)
+
+	certRotateCmd.PersistentFlags().BoolVar(&sshFlag.automate, "automate", false, "Automate ha server name to ssh")
+	certRotateCmd.PersistentFlags().BoolVar(&sshFlag.automate, "a2", false, "Automate ha server name to ssh")
+	certRotateCmd.PersistentFlags().BoolVar(&sshFlag.postgres, "postgres", false, "Postgres server name to ssh")
+	certRotateCmd.PersistentFlags().BoolVar(&sshFlag.postgres, "pg", false, "Postgres server name to ssh")
+
+	certRotateCmd.PersistentFlags().StringVar(&certFlags.rootCa, "root-ca", "", "Root CA value")
+	certRotateCmd.PersistentFlags().StringVar(&certFlags.privateCert, "private-cert", "", "Ha private certificate")
+	certRotateCmd.PersistentFlags().StringVar(&certFlags.publicCert, "public-cert", "", "Ha public certificate")
+}
+
+func certRotate(cmd *cobra.Command, args []string) error {
+	rootCaPath := certFlags.rootCa
+	privateCertPath := certFlags.privateCert
+	publicCertPath := certFlags.publicCert
+	fileName := "cert.toml"
+
+	if privateCertPath == "" || publicCertPath == "" {
+		return errors.New("Please provide public and private cert paths")
+	}
+
+	var rootCA []byte
+	var err error
+	if sshFlag.postgres {
+		if rootCaPath == "" {
+			return errors.New("Please provide rootCA path")
+		}
+		rootCA, err = ioutil.ReadFile(rootCaPath) // nosemgrep
+		if err != nil {
+			return status.Wrap(
+				err,
+				status.FileAccessError,
+				fmt.Sprintf("failed reading data from file: %s", err),
+			)
+		}
+	}
+
+	privateCert, err := ioutil.ReadFile(privateCertPath) // nosemgrep
+	if err != nil {
+		return status.Wrap(
+			err,
+			status.FileAccessError,
+			fmt.Sprintf("failed reading data from file: %s", err),
+		)
+	}
+
+	publicCert, err := ioutil.ReadFile(publicCertPath) // nosemgrep
+	if err != nil {
+		return status.Wrap(
+			err,
+			status.FileAccessError,
+			fmt.Sprintf("failed reading data from file: %s", err),
+		)
+	}
+
+	f, err := os.Create(fileName)
+	if err != nil {
+		log.Fatal(err)
+	}
+	infra, err := getAutomateHAInfraDetails()
+	if err != nil {
+		return err
+	}
+	sshUser := infra.Outputs.SSHUser.Value
+	sskKeyFile := infra.Outputs.SSHKeyFile.Value
+	sshPort := infra.Outputs.SSHPort.Value
+
+	if isA2HARBFileExist() {
+		if sshFlag.automate {
+			config := fmt.Sprintf(`
```

Comment:
Along with the cert changes, make sure to added all other applied configs to the .toml. Else all the modified values will overridden to default on service restart